src/asynk/async_worker.rs

```diff
@@ -88,14 +94,33 @@ where
 
     pub async fn run_tasks(&mut self) -> Result<(), Error> {
         loop {
+            //fetch task
             match self
                 .queue
                 .fetch_and_touch_task(Some(self.task_type.clone()))
                 .await
             {
                 Ok(Some(task)) => {
+                    let actual_task: Box<dyn AsyncRunnable> =
+                        serde_json::from_value(task.metadata.clone()).unwrap();
+
+                    // check if task is scheduled or not
                     self.sleep_params.maybe_reset_sleep_period();
-                    self.run(task).await?
+
+                    match actual_task.cron() {
+                        None => self.run(task, actual_task).await?,
+                        Some(scheduled) => {
+                            match scheduled {
+                                CronPattern(_) => {
+                                    // program task
+                                    self.queue.schedule_task(&*actual_task).await?;
+                                    // run scheduled task
+                                    self.run(task, actual_task).await?;
+                                }
+                                ScheduleOnce(_) => self.run(task, actual_task).await?,
```

Comment:
`run` can also be outside of this match

Reply:
[Comments adressed](https://github.com/ayrat555/fang/pull/71/commits/e32b0fe5b77adb1ff9ec8a9453335b37a00bfccc)